server/src/main/java/org/opensearch/tasks/TaskCancelledException.java

```diff
@@ -42,12 +45,38 @@
  * @opensearch.internal
  */
 public class TaskCancelledException extends OpenSearchException {
+    private final RestStatus restStatus;
```

Comment:
Instead of modifying `TaskCancelledException` can we not throw an overridden exception `ResourceLimitExeceededException` with status 429